pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -0,0 +1,209 @@
+	//if ingress exists, update endpoints if different
+
+	var isUpdate bool
+
+	//check if ingress for this globaldns is already present
+	ingress, err := n.getIngressForGlobalDNS(obj)
+
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return nil, fmt.Errorf("GlobalDNSController: Error listing ingress for the GlobalDNS %v", err)
+	}
+
+	if ingress != nil && err == nil {
+		isUpdate = true
+	}
+
+	if len(obj.Status.Endpoints) == 0 && !isUpdate {
+		return nil, nil
+	}
+
+	if !isUpdate {
+		ingress, err = n.createIngressForGlobalDNS(obj)
+		if err != nil {
+			return nil, fmt.Errorf("GlobalDNSController: Error creating an ingress for the GlobalDNS %v", err)
+		}
+	}
+
+	err = n.updateIngressEndpoints(ingress, obj.Status.Endpoints)
+	if err != nil {
+		return nil, fmt.Errorf("GlobalDNSController: Error updating ingress for the GlobalDNS %v", err)
+	}
+
+	return nil, nil
+}
+
+func (n *GDController) getIngressForGlobalDNS(globaldns *v3.GlobalDNS) (*v1beta1.Ingress, error) {
+	ingress, err := n.ingresses.Get(strings.Join([]string{"globaldns-ingress", globaldns.Name}, "-"), metav1.GetOptions{}) //n.Get("", strings.Join([]string{"globaldns-ingress", globaldns.Name}, "-"))
+	if err != nil {
+		return nil, err
+	}
+	//make sure the ingress is owned by this globalDNS
+	if n.isIngressOwnedByGlobalDNS(ingress, globaldns) {
+		return ingress, nil
+	}
+	return nil, nil
+}
+
+func (n *GDController) isIngressOwnedByGlobalDNS(ingress *v1beta1.Ingress, globaldns *v3.GlobalDNS) bool {
+	for i, owners := 0, ingress.GetOwnerReferences(); owners != nil && i < len(owners); i++ {
+		if owners[i].UID == globaldns.UID && owners[i].Kind == globaldns.Kind {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *GDController) createIngressForGlobalDNS(globaldns *v3.GlobalDNS) (*v1beta1.Ingress, error) {
+	ingressSpec := n.generateNewIngressSpec(globaldns)
+	ingressObj, err := n.ingresses.Create(ingressSpec)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Created ingress %v for globalDNS %s", ingressObj.Name, globaldns.Name)
+	return ingressObj, nil
+}
+
+func (n *GDController) generateNewIngressSpec(globaldns *v3.GlobalDNS) *v1beta1.Ingress {
+	controller := true
+	return &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: strings.Join([]string{"globaldns-ingress", globaldns.Name}, "-"),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Name:       globaldns.Name,
+					APIVersion: "v3",
+					UID:        globaldns.UID,
+					Kind:       globaldns.Kind,
+					Controller: &controller,
+				},
+			},
+			Namespace: globaldns.Namespace,
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: globaldns.Spec.FQDN,
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "http-svc-dummy",
+										ServicePort: intstr.IntOrString{
+											Type:   intstr.Int,
```

Comment:
the code formatting is funny here :) do you think we can put all }} on one line?